lib/components/map/default-map.js

```diff
@@ -162,43 +258,55 @@ class DefaultMap extends Component {
           <RouteViewerOverlay />
           <TransitVehicleOverlay />
           <StopViewerOverlay />
-          <TransitiveOverlay getTransitiveRouteLabel={getTransitiveRouteLabel} />
+          <TransitiveOverlay
+            getTransitiveRouteLabel={getTransitiveRouteLabel}
+          />
           <TripViewerOverlay />
           <ElevationPointMarker />
 
           {/* The configurable overlays */}
-          {mapConfig.overlays && mapConfig.overlays.map((overlayConfig, k) => {
+          {mapConfig.overlays?.map((overlayConfig, k) => {
+            const namedLayerProps = {
+              ...overlayConfig,
+              key: k,
+              name:
+                overlayConfig.name || getLayerName(overlayConfig, config, intl)
```

Comment:
Would it be clearer what's going on if we put the `overlayConfig.name` inside `getLayerName` and replaced the `.name` default case?

Reply:
Good one! Refactored in https://github.com/opentripplanner/otp-react-redux/pull/606/commits/1162dc3bc6d0532c5ef6ad387f9c61d193527718.